command/server/config.go

```diff
@@ -905,3 +905,120 @@ func parseTelemetry(result *Config, list *ast.ObjectList) error {
 
 	return nil
 }
+
+// Sanitized returns a copy of the config with all values that are considered
+// sensitive stripped. It also strips all raw values that are mainly
+// used for parsing.
+//
+// Specifically, the fields that this method strips are:
+// - Storage.Config
+// - HAStorage.Config
+// - Seals.Config
+// - Telemetry.CirconusAPIToken
+func (c *Config) Sanitized() *Config {
+	// Sanitize storage stanza
+	var sanitizedStorage *Storage
+	if c.Storage != nil {
+		sanitizedStorage = &Storage{
+			Type:              c.Storage.Type,
+			RedirectAddr:      c.Storage.RedirectAddr,
+			ClusterAddr:       c.Storage.ClusterAddr,
+			DisableClustering: c.Storage.DisableClustering,
+		}
+	}
+
+	// Sanitize HA storage stanza
+	var sanitizedHAStorage *Storage
+	if c.HAStorage != nil {
+		sanitizedHAStorage = &Storage{
+			Type:              c.HAStorage.Type,
+			RedirectAddr:      c.HAStorage.RedirectAddr,
+			ClusterAddr:       c.HAStorage.ClusterAddr,
+			DisableClustering: c.HAStorage.DisableClustering,
+		}
+	}
+
+	// Sanitize seals stanza
+	var sanitizedSeals []*Seal
+	if len(c.Seals) != 0 {
+		for _, s := range c.Seals {
+			cleanSeal := &Seal{
+				Type:     s.Type,
+				Disabled: s.Disabled,
+			}
+			sanitizedSeals = append(sanitizedSeals, cleanSeal)
+		}
+	}
+
+	// Sanitize telemetry stanza
+	var sanitizedTelemetry *Telemetry
+	if c.Telemetry != nil {
+		sanitizedTelemetry = &Telemetry{
+			StatsiteAddr:                       c.Telemetry.StatsiteAddr,
+			StatsdAddr:                         c.Telemetry.StatsdAddr,
+			DisableHostname:                    c.Telemetry.DisableHostname,
+			CirconusAPIToken:                   "",
```

Comment:
Wouldn't it be easier to copy c.Telemetry and then overwrite CirconusAPIToken with `""`?  That would also save us having to update this code any time we add a new field.

Reply:
My concern of doing it this way is that we could accidentally output any newly added sensitive fields if we forget to overwrite it here.